### PR TITLE
Follow Vault convention on `DELETE` being idempotent

### PR DESCRIPTION
**Pull request by jefferai**
with audit/auth/mounts deletes (a.k.a. disabling/unmounting).

### CODE REVIEW

File: vault/auth.go [hunk 1]
```diff
@@ -100,63 +100,64 @@ func (c *Core) enableCredential(entry *MountEntry) error {
...
+func (c *Core) disableCredential(path string) (bool, error) {
 	// Ensure we end the path in a slash
 	if !strings.HasSuffix(path, "/") {
 		path += "/"
 	}
 
 	// Ensure the token backend is not affected
 	if path == "token/" {
-		return fmt.Errorf("token credential backend cannot be disabled")
+		return true, fmt.Errorf("token credential backend cannot be disabled")
```
Comment (vishalnayak) [inline vault/auth.go]: Here, we'll not know if the backend existed or not. No?

Comment (jefferai) [inline vault/auth.go]: Yep, that's why it returns true.

**Comment by vishalnayak** [comment]
For secret deletions, there was an indicator added to the CLI output (Success! Deleted 'secret/hello' if it existed). May be a good thing to add "if it existed" in for the mounts as well?

**Comment by jefferai** [comment]
@vishalnayak So the reason I ran across this was I was doing a `DELETE` against `sys/auth/[endpoint]` which would indeed give that indicator. But that comes from the CLI, so yes -- we can add to the CLI for the specific extra commands.

File: vault/auth.go (hunk 1 above)
Comment (vishalnayak) [inline vault/auth.go]: Okay and yes, the behavior seems correct. Ignore this comment.

**Pull request merged**